Mathlib/Algebra/Homology/ShortComplex/LeftHomology.lean

```diff
@@ -61,6 +60,206 @@ structure LeftHomologyData where
 
 initialize_simps_projections LeftHomologyData (-hi, -hπ)
 
+namespace LeftHomologyData
+
+/-- the chosen kernels and cokernels of the limits API give a `LeftHomologyData` -/
+@[simps]
+noncomputable def ofHasKernelOfHasCokernel
+    [HasKernel S.g] [HasCokernel (kernel.lift S.g S.f S.zero)] :
+  S.LeftHomologyData where
+  K := kernel S.g
+  H := cokernel (kernel.lift S.g S.f S.zero)
+  i := kernel.ι _
+  π := cokernel.π _
+  wi := kernel.condition _
+  hi := kernelIsKernel _
+  wπ := cokernel.condition _
+  hπ := cokernelIsCokernel _
+
+attribute [reassoc (attr := simp)] wi wπ
+
+variable {S}
+variable (h : S.LeftHomologyData) {A : C}
+
+instance : Mono h.i := ⟨fun _ _ => Fork.IsLimit.hom_ext h.hi⟩
+
+instance : Epi h.π := ⟨fun _ _ => Cofork.IsColimit.hom_ext h.hπ⟩
+
+/-- any morphism `k : A ⟶ S.X₂` that is a cycle (i.e. `k ≫ S.g = 0`) lifts
```

Comment:
I notice that some of your docstrings start with a lowercase letter, others with uppercase. Is there a pattern?
Anyway, I think I prefer that all docstrings start with an uppercase letter.

Reply:
Ok, I have changed to Uppercase for the definitions, but kept lowercase for docstrings of fields of structures.